plugins/module_utils/identity/keycloak/keycloak.py

```diff
@@ -747,8 +756,15 @@ def get_user_by_username(self, username, realm="master"):
         users_url = URL_USERS.format(url=self.baseurl, realm=realm)
         users_url += '?username=%s&exact=true' % username
         try:
-            return json.loads(to_native(open_url(users_url, method='GET', headers=self.restheaders, timeout=self.connection_timeout,
-                                                 validate_certs=self.validate_certs).read()))
+            userrep = None
+            users = json.loads(to_native(open_url(users_url, method='GET', headers=self.restheaders, timeout=self.connection_timeout,
+                                                  validate_certs=self.validate_certs).read()))
+            for user in users:
+                if user['username'] == username:
+                    userrep = user
+                    break
+            return userrep
```

Comment:
Is this a bugfix? Or does this change behavior of this existing function?

Reply:
In regards of the function documentation and it's usage in the keycloak_user_rolemapping module, this is a bug fix.
The original code returned an array of users instead of the user representation.
 def get_user_by_username(self, username, realm="master"):
        """ Fetch a keycloak user within a realm based on its username.

        If the user does not exist, None is returned.
        :param username: Username of the user to fetch.
        :param realm: Realm in which the user resides; default 'master'
        """

Reply:
Could you please separate this bugfix into another PR so we can backport it? Bugfixes and features shouldn't be mixed in the same PR if possible.

Reply:
Ping ^

Reply:
Here is the PR for this bug fix: https://github.com/ansible-collections/community.general/pull/6568

Reply:
Can you rebase to the current `main` branch so this part disappears from this PR?